### PR TITLE
IPC pipe

### PR DESCRIPTION
**Pull request by cprovencher**
the backtester can now communicate simulation results to the backend

### CODE REVIEW

**Review by anadeau1 — APPROVED** [review]

File: client/components/Controls.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ class Controls extends React.Component {
   launchBacktest() {
     // TODO: unify saveTemplate and launchBacktest
     if (this.checkForErrors()) return;
-    api.post('backtests', {
```
Comment (mrancourt) [inline client/components/Controls.js:78]: I think `backtester/run` or even `backtester/new` would be more REST-ish

File: server/app.py [hunk 2]
```diff
@@ -47,5 +49,16 @@ def quote(id):
     quote = Quote.query.get(id)
     return jsonify(quote.serialize)
 
+@app.route('/run_backtester')
```
Comment (mrancourt) [inline server/app.py]: again, `/backtester/run` ?

File: server/app.py [hunk 3]
```diff
@@ -47,5 +49,16 @@ def quote(id):
...
+def run_backtester(algorithm, params, tickers):
```
Comment (mrancourt) [inline server/app.py]: Does that work? I think you would need to allow POST with `methods=['POST']` and then get the POST parameters with the Flask `request` helper. 

Something like:
```python
from flask import request
#...
@app.route('/backtester/run', methods=['POST'])
    def run_backtester():
        data = request.get_json()
        #...
```

**Review by mrancourt — APPROVED** [review]

File: server/app.py [hunk 4]
```diff
@@ -59,16 +61,20 @@ def quote(id):
...
-    line = ''
-    while line != 'BACKTESTER DONE':
+@app.route('/backtester/run', methods=['POST'])
+def run_backtester():
+    post_data = request.get_json()
+    params = "".join(str(post_data['params']).split())
+    exe = os.path.dirname(os.path.abspath(__file__)) + '/../backtester/backtester/Release/backtester.exe'
+    proc = subprocess.Popen([exe, '--algoName', post_data['algorithm'], '--params', params, '--tickers'] +
+                            post_data['tickers'], stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+    while 1:
```
Comment (mrancourt) [inline server/app.py:71]: Why not 
```python
while line != 'BACKTESTER DONE'
```

**Pull request merged**